File: Tutorial_Sample_Code/ViewerObjectModel/FilterData/CS/CS_Win_CRVObjMod_FilteringData/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Shared;

namespace CS_Win_CRVObjMod_FilteringData
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void ConfigureCrystalReports()
        {
            selectOperatorList.DataSource = System.Enum.GetValues(typeof(CeComparisonOperator));

            string selectFormula = "{Customer.Last Year's Sales} > 11000.00 "
 + "AND Mid({Customer.Customer Name}, 1, 1) = \"A\"";
            crystalReportViewer.SelectionFormula = selectFormula;

            string reportPath = Application.StartupPath + "\\" + "CustomersBySalesName.rpt";
            crystalReportViewer.ReportSource = reportPath;
        }


        private void Form1_Load(object sender, EventArgs e)
        {
            ConfigureCrystalReports();
        }

        private void crystalReportViewer_Load(object sender, EventArgs e)
        {

        }

        private void redisplay_Click(object sender, EventArgs e)
        {
            string selectedOperator = GetSelectedCompareOperator();

            string selectFormula = "{Customer.Last Year's Sales} > " + lastYearsSales.Text
+ " AND Mid({Customer.Customer Name}, 1, 1) " + selectedOperator + " \"" + customerName.Text + "\"";

            crystalReportViewer.SelectionFormula = selectFormula;
            crystalReportViewer.ReportSource = Application.StartupPath + "\\" + "CustomersBySalesName.rpt";
        }

        private string GetSelectedCompareOperator()
        {
            switch ((CeComparisonOperator)selectOperatorList.SelectedIndex)
            {
                case CeComparisonOperator.EqualTo:
                    return "=";
                case CeComparisonOperator.LessThan:
                    return "<";
                case CeComparisonOperator.GreaterThan:
                    return ">";
                case CeComparisonOperator.LessThan_or_EqualTo:
                    return "<=";
                case CeComparisonOperator.GreaterThan_or_EqualTo:
                    return ">=";
                case CeComparisonOperator.Not_EqualTo:
                    return "<>";
                default:
                    return "=";
            }
        }

    }
}

File: Tutorial_Sample_Code/ViewerObjectModel/FilterData/VB/VB_Win_CRVObjMod_FilteringData/Form1.vb
Imports CrystalDecisions.CrystalReports.Engine
Imports CrystalDecisions.Shared

Public Class Form1

    Private Sub ConfigureCrystalReports()
        selectOperatorList.DataSource = System.Enum.GetValues(GetType(CeComparisonOperator))

        Dim mySelectFormula As String = "{Customer.Last Year's Sales} > 11000.00 " _
        & "AND Mid({Customer.Customer Name}, 1, 1) = ""A"" "
        myCrystalReportViewer.SelectionFormula = mySelectFormula

        Dim reportPath As String = Application.StartupPath & "\" & "CustomersBySalesName.rpt"
        myCrystalReportViewer.ReportSource = reportPath


    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ConfigureCrystalReports()
    End Sub

    Private Sub redisplay_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles redisplay.Click
        Dim mySelectedOperator As String = GetSelectedCompareOperator()

        Dim mySelectFormula As String = "{Customer.Last Year's Sales} > " & lastYearsSales.Text _
& " AND Mid({Customer.Customer Name}, 1, 1) " & mySelectedOperator & " """ & customerName.Text & """"
        myCrystalReportViewer.SelectionFormula = mySelectFormula
        myCrystalReportViewer.ReportSource = Application.StartupPath & "\" & "CustomersBySalesName.rpt"
    End Sub

    Private Function GetSelectedCompareOperator() As String
        Select Case selectOperatorList.SelectedIndex
            Case CeComparisonOperator.EqualTo
                Return "="
            Case CeComparisonOperator.LessThan
                Return "<"
            Case CeComparisonOperator.GreaterThan
                Return ">"
            Case CeComparisonOperator.LessThan_or_EqualTo
                Return "<="
            Case CeComparisonOperator.GreaterThan_or_EqualTo
                Return ">="
            Case CeComparisonOperator.Not_EqualTo
                Return "<>"
            Case Else
                Return "="
        End Select

    End Function

End Class
